pkg/client/client.go

```diff
@@ -632,29 +649,106 @@ func (c *Client) GetAlertingRule(ctx context.Context, namespace, name string) (*
 	return c.osmclient.MonitoringV1().AlertingRules(namespace).Get(ctx, name, metav1.GetOptions{})
 }
 
-func (c *Client) CreateOrUpdatePrometheus(ctx context.Context, p *monv1.Prometheus) error {
-	pclient := c.mclient.MonitoringV1().Prometheuses(p.GetNamespace())
-	existing, err := pclient.Get(ctx, p.GetName(), metav1.GetOptions{})
+func (c *Client) CreateOrUpdatePrometheus(ctx context.Context, structuredRequiredPrometheus *monv1.Prometheus) (*bool, error) {
```

Comment:
why do we need to return a pointer instead of just a bool here?
I don't think tests should rely on this, I think we should retrieve whether the object was updated from the API/client.

Reply:
I looked at the usages and there doesn't seem to be a need for three states. Fixed! 👍🏼

Reply:
I looked at the usages and there doesn't seem to be a need for three states. Fixed! 👍🏼